src/ui/components/NetworkMonitor/RequestTable.tsx

```diff
@@ -30,14 +42,65 @@ const RequestTable = ({
 }) => {
   const { columns, getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = table;
 
+  const dispatch = useDispatch();
   const loadedRegions = useSelector(getLoadedRegions);
+  const focusRegion = useSelector(getFocusRegion);
+  const [contextMenuData, setContextMenuData] = useState<ContextMenuData | null>(null);
 
   const onSeek = (request: RequestSummary) => {
     trackEvent("net_monitor.seek_to_request");
     seek(request.point.point, request.point.time, true);
     onRowSelect(request);
   };
 
+  const closeContextMenu = () => {
+    setContextMenuData(null);
+  };
+
+  const setFocusEnd = () => {
+    setContextMenuData(null);
+
+    if (contextMenuData) {
+      const time = contextMenuData.row.original?.end;
+      if (time != null) {
+        // If this is the first time the user is focusing, start at the beginning of the recording (or zoom region).
```

Comment:
What do you think about DRYing this up with the same section in `ConsoleOutput`?

Reply:
Sure. I'll put a method in `utils/time` and also move the point-to-time lookup function there too.